src/pytorch_lightning/tuner/batch_size_scaling.py

```diff
@@ -234,18 +232,27 @@ def _adjust_batch_size(
     """
     model = trainer.lightning_module
     batch_size = lightning_getattr(model, batch_arg_name)
-    new_size = value if value is not None else int(batch_size * factor)
+    if value is not None:
+        new_size = value
+    else:
+        if not isinstance(batch_size, int):
+            raise ValueError(f"Batch size attribute in LightningModule must be an integer, got: {batch_size!r}")
+        new_size = int(batch_size * factor)
```

Comment:
```suggestion
    new_size: int = value if value is not None else int(batch_size * factor)
```
that doesn't work?

Reply:
`batch_size` may be None. So mypy says error about `*` not supporting with `None` and `float` .